### ISSUE TITLE
Bad default for multivariate normal

### ISSUE THREAD

**Issue by andreasnoack** [JuliaStats/Distributions.jl#657]
```julia
julia> D = MvNormal(zeros(2))
ZeroMeanDiagNormal(
dim: 2
μ: [0.0, 0.0]
Σ: [0.0 0.0; 0.0 0.0]
)
```

**Comment by simonbyrne** [comment]
Yes, let's deprecate `MvNormal(::AbstractVector)` and replace it with `MvNormal(::Diagonal)`

**Comment by andreasnoack** [comment]
Oh. It didn't even cross my mind that it would be the variance. I assumed that a vector would be the mean. So yes, let's definitely deprecate that for a `Diagonal` version. Eventually, I think we could consider reintroducing the constructor where the vector specifies the mean.

**Comment by simonbyrne** [comment]
Actually, even more confusingly, it's standard deviation.
